pkg/network/ebpf/c/protocols/protocol-classification-helpers.h

```diff
@@ -57,6 +69,51 @@ static __always_inline bool is_http2(const char* buf, __u32 buf_size) {
     return is_http2_preface(buf, buf_size) || is_http2_server_settings(buf, buf_size);
 }
 
+// The method checks if the given buffer starts is amqp message.
+// Ref: https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf
+static __always_inline bool is_amqp(const char* buf, __u32 buf_size) {
+    if (buf_size < AMQP_MIN_FRAME_LENGTH) {
+        return false;
+    }
+
+    // New connection should start with protocol header of AMQP.
+    // Ref https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf.
+    if (is_amqp_protocol_header(buf, buf_size)) {
+        return true;
+    }
+
+    uint8_t frame_type = buf[0];
+    // Check only for method frame type.
+    if (frame_type != AMQP_FRAME_METHOD_TYPE) {
+        return false;
+    }
+
+    __u16 class_id = buf[7] << 8 | buf[8];
+    __u16 method_id = buf[9] << 8 | buf[10];
```

Comment:
explain what are we doing here, and why

Reply:
furthermore, you checked the buffer size if not less than `AMQP_MIN_FRAME_LENGTH` (8), but here you are accessing the positions `8`, `9` & `10` which exceed from the length you verified

Reply:
the min frame length is indeed 8 which may include header and a payload, if we got to the part which we extract the class id and the method id it is definitely part of the "method frame" which should be bigger then 8 for sure and we should be able to extract those fields

Reply:
I meant that you are right, in the protocol the minimum message can be 8 bytes, but here, you are assuming that the positions `8`, `9` & `10` are valid to access in the buffer, so you are expecting to have payloads with length of at least 11 characters.